src/components/hub/elements/MessagesList.js

```diff
@@ -114,6 +117,25 @@ export default function MessagesList({ scrollRef }) {
     }, 300);
   }, []);
 
+  useEffect(() => {
+    const usersToUpdate = new Set();
+
+    messages.forEach((msg) => {
+      if (!participants[msg.from]) {
+        usersToUpdate.add(msg.from);
+      }
+      if (msg.x?.user?._id && !participants[msg.x.user._id]) {
+        usersToUpdate.add(msg.x.user._id);
+      }
+    });
+
+    if (usersToUpdate.size) {
+      api
+        .getUsersByIds({ ids: [...usersToUpdate] })
+        .then((users) => dispatch(addUsers(users)));
+    }
+  }, []);
```

Comment:
what will happen if a user scrolls to load more? will these older messages be covered as well?